api/middleware.go.tmpl

```diff
@@ -52,8 +52,9 @@ func TokenMiddleware(psk []byte, public map[string]string, h http.Handler) http.
 			log.Infof("Authenticating token for protected URL '%s'", r.URL)
 
 			htoken := r.Header.Get("X-Auth-Token")
-			if err := bcrypt.CompareHashAndPassword([]byte(htoken), psk); err != nil {
-				log.Warnf("Unable to authenticate session for '%s' with '%s'", r.URL, htoken)
+			if err := bcrypt.CompareHashAndPassword(psk, []byte(htoken)); err != nil {
+				log.Warnf("Unable to authenticate session for url: '%s', with error: '%s'", r.URL, err)
+				log.Debugf("Unable to authenticate session for url: '%s', with error: '%s', with user input: '%s'", r.URL, err, htoken)
```

Comment:
I think you can add this to an API if you're debugging something.  It doesn't need to be in the middleware.

Reply:
if you want to just change the warning message to have the error and not the `htoken`, that's probably better although bcrypt is a one way hash, so theoretically it should be fairly safe to include it.

Reply:
yeah, I guess it's well enough understood in the middleware as-is.  I did drop the htoken.  A log.Debugf statement can be added by the implementer of the template as you stated to debug inputs, etc.